core-bundle/src/Resources/contao/library/Contao/Database/Result.php

```diff
@@ -392,10 +392,15 @@ public function row($blnEnumerated=false)
 
 		if (!$this->isModified)
 		{
+			if (!isset($this->resultSet[$this->intIndex]))
+			{
+				return array();
+			}
+
 			return $blnEnumerated ? array_values($this->resultSet[$this->intIndex]) : $this->resultSet[$this->intIndex];
 		}
 
-		$row = array_merge($this->resultSet[$this->intIndex], $this->arrModified);
+		$row = array_merge(isset($this->resultSet[$this->intIndex]) ? $this->resultSet[$this->intIndex] : array(), $this->arrModified);
```

Comment:
```suggestion
		$row = $this->arrModified;

		if (isset($this->resultSet[$this->intIndex]))
		{
			$row = array_merge($this->resultSet[$this->intIndex], $this->arrModified);
		}
```

This way we would not call array_merge() at all if `$this->resultSet[$this->intIndex]` is not set. Not sure if that makes a difference though.

Reply:
I think in 99% of the cases `$this->arrModified` is empty and `$this->resultSet[$this->intIndex]` is set so I would prefer the current way which does not copy the `$this->arrModified` array.